WordPress/gutenberg
# Move and refactor Inserter component as reused in header tools

This pull request seeks to apply a few refactorings to the `<Inserter />` component:

- Move from `editor/inserter` to `editor/components/inserter`
- Change entry point from `components/inserter/button` to `components/inserter`
  - Should enforce a general preference toward `index.js` as entry point of any component. If we want menu to be a separate independent component, we should move it to its own directory to reflect as such, but currently we don't have a need for this.
- Render `<Inserter />` in Header tools

__Testing Instructions:__

Verify that an inserter button is shown both in the header and at the bottom of post content and can be toggled to display a menu overlay.

__Future Tasks:__

- Header inserter overlay position should open downward, not upward
- To the first point, we may want to consider extracting overlay/tooltip logic to a separate component (perhaps with logic detecting adapting to display in opposite direction when desired opening direction exceeds viewport bounds)
